### ISSUE TITLE
Can you provide the code for Logg3D's EVAL?

### ISSUE THREAD

**Issue by Stream11** [csiro-robotics/Wild-Places#12]
Thank you for your excellent work.  i try to use wild-place on  logg3d, I'm having trouble modifying the code。logg3d eval  ，There is no adaptation to the dataset。
![image](https://github.com/user-attachments/assets/49b258ef-b70a-40b4-bbea-279a310e81af)

**Comment by kavisha725** [comment]
Use the utility function provided [here](https://github.com/csiro-robotics/Wild-Places/blob/eafde14a4a1aeb5e96a5d56c12ed046bcfdb02d9/scripts/eval/logg3d/logg3d_utils.py#L76) to generate latent_vectors for `database_features` and `query_features`. 
Then, use the evaluation instructions provided [here](https://github.com/csiro-robotics/Wild-Places/tree/main?tab=readme-ov-file#33-evaluation).

**Comment by kavisha725** [comment]
Closing this issue for now. Feel free to reopen if you have more questions.

**Comment by Stream11** [comment]
> Closing this issue for now. Feel free to reopen if you have more questions.暂时关闭此问题。如果您有更多问题，请随时重新打开。

Thank you for your help. I have encountered a new issue where the feature dimension output by logg3d is 256, but the default feature dimension of get_latest_vectors is 1024. Do I need to set gd-dim to 256?

**Comment by kavisha725** [comment]
As suggested in the instructions [here](https://github.com/csiro-robotics/Wild-Places/blob/eafde14a4a1aeb5e96a5d56c12ed046bcfdb02d9/scripts/eval/logg3d/logg3d_utils.py#L78), the LoGG3D checkpoint for the WildPlaces project was trained with a local feature dimension of 32, which will result in a global feature dimension of 1024.

**Comment by Stream11** [comment]
> As suggested in the instructions [here](https://github.com/csiro-robotics/Wild-Places/blob/eafde14a4a1aeb5e96a5d56c12ed046bcfdb02d9/scripts/eval/logg3d/logg3d_utils.py#L78), the LoGG3D checkpoint for the WildPlaces project was trained with a local feature dimension of 32, which will result in a global feature dimension of 1024.如此[处](https://github.com/csiro-robotics/Wild-Places/blob/eafde14a4a1aeb5e96a5d56c12ed046bcfdb02d9/scripts/eval/logg3d/logg3d_utils.py#L78)的说明所示，WildPlaces 项目的 LoGG3D 检查点是使用 32 的局部特征维度进行训练的，这将导致全局特征维度为 1024。

Thank you for your patient answers

**Comment by Stream11** [comment]
Thank you for your help.
I reproduced the intra-sequency.py from K-04, but the results are not good. I believe there might be an issue with my training parameters or input data. Could you give me some advice?

I used the Clouds_downsampled point cloud from K-04 and trained it using training_wild-places.pickle generated by training_set.py, while using K-04.pickle from testing_sets.py as validation input. Is there any problem with this?
Do you have any suggestions for the parameter settings in train_config.py like downsample, gp_vals, base_learning_rate, etc.?

**Comment by kavisha725** [comment]
- First, were you able to recreate the results in the paper using the publicly released checkpoint?
- For training, try increasing the `negatives_per_query` [here](https://github.com/csiro-robotics/LoGG3D-Net/blob/cfbe064e36a4c39ef1bf6bb2d0304c6f9c7a0905/config/train_config.py#L34).

**Comment by Stream11** [comment]
> * First, were you able to recreate the results in the paper using the publicly released checkpoint?首先，您是否能够使用公开发布的检查点在论文中重现结果？
> * For training, try increasing the `negatives_per_query` [here](https://github.com/csiro-robotics/LoGG3D-Net/blob/cfbe064e36a4c39ef1bf6bb2d0304c6f9c7a0905/config/train_config.py#L34).对于培训，请尝试[在此处](https://github.com/csiro-robotics/LoGG3D-Net/blob/cfbe064e36a4c39ef1bf6bb2d0304c6f9c7a0905/config/train_config.py#L34)增加 `negatives_per_query`。
Yes, I was able to recreate the results in the paper using the publicly released checkpoint.
Thank you for your help.

**Comment by Stream11** [comment]
> * First, were you able to recreate the results in the paper using the publicly released checkpoint?首先，您是否能够使用公开发布的检查点在论文中重现结果？
> * For training, try increasing the `negatives_per_query` [here](https://github.com/csiro-robotics/LoGG3D-Net/blob/cfbe064e36a4c39ef1bf6bb2d0304c6f9c7a0905/config/train_config.py#L34).对于培训，请尝试[在此处](https://github.com/csiro-robotics/LoGG3D-Net/blob/cfbe064e36a4c39ef1bf6bb2d0304c6f9c7a0905/config/train_config.py#L34)增加 `negatives_per_query`。

Which sequences are the training datasets used for this checkpoint. thank you !

**Comment by kavisha725** [comment]
Training details are given in Section V(A) in the [paper](https://arxiv.org/pdf/2211.12732).

**Comment by Stream11** [comment]
> Training details are given in Section V(A) in the [paper](https://arxiv.org/pdf/2211.12732).培训详情在[论文](https://arxiv.org/pdf/2211.12732)的 V（A） 节中给出。

k-04  Intra-Sequence Evaluation， training using the data from Section V(A), or only use the data from k-04?  thank you !
